### PR TITLE
Document function of stale bot in handling pull request states

### PR DESCRIPTION
**Pull request by harmut01**
## Summary 
Add documentation explaining how the stale bot works and how it fits into how we manage pull requests.

### CODE REVIEW

File: docs/contributing/guidelines/workflow.md [hunk 1]
```diff
@@ -204,9 +204,22 @@ Each pull request goes through the following workflow:
 
 Mergify bot drives our workflow. The mergify rules are defined in the Mbed OS repository in the .mergify.yml file. The Mbed OS maintainers are responsible for moving pull requests through the workflow states with help from the mergify bot.
 
-Each state is time-boxed. In most cases, this is sufficient time to move to another state. The pull request can be closed if no update is provided within the time frame.
+Each state is time-boxed. In most cases, sufficient time is provided to move to another state. The ciarmcom bot periodically checks that pending activities on pull requests are completed in a timely manner. 
+<center>
 
-If a pull request is idle for more than two weeks, it will be closed. The author or the maintainer can reopen it at any time.
+| State           | Limit (days)  |
```
Comment (0xc0170) [inline docs/contributing/guidelines/workflow.md:210]: the time should be "working-days", is it implemented that way ? Otherise CI/ready for merge Prs become stale during weekends.

File: docs/contributing/guidelines/workflow.md [hunk 2]
```diff
@@ -204,9 +204,22 @@ Each pull request goes through the following workflow:
...
+| --------------- |:-------------:|
+| needs: work     | 10            |
+| needs: review   | 3             |
+| needs: CI       | 1             |
+| ready for merge | 1             |
+
+</center>
+
+
+If the pull request is idle in a state for longer than the allowed time, our bot will add the 'stale' label alongside a comment nudging the user to carry out any necessary work. This label will only be removed when the action required to move the pull request forward is completed.
```
Comment (0xc0170) [inline docs/contributing/guidelines/workflow.md]: not only a user, also reviewers are responsible

File: docs/contributing/guidelines/workflow.md (hunk 1 above)
Comment (harmut01) [reply docs/contributing/guidelines/workflow.md:210]: No, this isn't the case at the moment. I need to raise an issue to do this and fix handling of 'ready for merge' state.

**Review by 0xc0170 — APPROVED** [review]
We will fix the weekend days exclusion.

**Pull request merged**